File: src/main/org/nlogo/api/ViewInterface.java
package org.nlogo.api ;

public interface ViewInterface
{
	boolean viewIsVisible() ;
	void framesSkipped() ;

	boolean isDead() ;
	void paintImmediately( boolean force ) ;
	void incrementalUpdateFromEventThread() ;
	void repaint() ;

	double mouseXCor() ;
	double mouseYCor() ;
	boolean mouseDown() ;
	boolean mouseInside() ; 
	void resetMouseCors() ;

	void shapeChanged( org.nlogo.api.Shape shape ) ;
	void applyNewFontSize( int fontSize , int zoom ) ;
}

File: netlogo-core/src/main/api/ViewInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

import org.nlogo.core.Shape

trait ViewInterface {
  def viewIsVisible: Boolean
  def framesSkipped(): Unit
  def isDead: Boolean
  def paintImmediately(force: Boolean): Unit
  def incrementalUpdateFromEventThread(): Unit
  def repaint(): Unit
  def mouseXCor: Double
  def mouseYCor: Double
  def mouseDown: Boolean
  def mouseInside: Boolean
  def resetMouseCors(): Unit
  def shapeChanged(shape: Shape): Unit
  def applyNewFontSize(fontSize: Int, zoom: Int): Unit
}
